test/fluree/http_api/system_test.clj

```diff
@@ -297,67 +309,120 @@
+                         "Accept"       "application/json"}
+          txn-req       {:body
+                         (json/write-value-as-string
+                          {"ledger" ledger-name
+                           "txn"    [{"id"      "ex:query-test"
+                                      "type"    "schema:Test"
+                                      "ex:name" "query-test"}]})
+                         :headers json-headers}
+          txn-res       (post :create txn-req)
+          _             (assert (= 201 (:status txn-res)))
+          txn2-req      {:body
+                         (json/write-value-as-string
+                          {"ledger" ledger-name
+                           "txn"    [{"id"           "ex:query-test"
+                                      "ex:test-type" "integration"}]})
+                         :headers json-headers}
+          txn2-res      (post :transact txn2-req)
+          _             (assert (= 200 (:status txn2-res)))
+          query-req     {:body
+                         (json/write-value-as-string
+                          {"ledger" ledger-name
+                           "query"  {"commit-details" true
+                                     "t"              {"at" "latest"}}})
+                         :headers json-headers}
+          query-res     (post :history query-req)
+          query-results (-> query-res :body json/read-value)]
       (is (= 200 (:status query-res))
           (str "History query response was: " (pr-str query-res)))
-      (is (= [{"f:commit"
-               {"f:data"
-                {"f:assert"
-                 [{"ex:name"  "query-test"
-                   "id"       "ex:query-test"
-                   "rdf:type" ["schema:Test"]}]
-                 "f:retract" []}}}]
-             (-> query-res :body json/read-value))))))
+      (let [history-expectations
+            {["id"]                  #(re-matches commit-id-regex %)
+             ["f:address"]           #(re-matches mem-addr-regex %)
+             ["f:alias"]             #(= % ledger-name)
+             ["f:branch"]            #(= % "main")
+             ["f:defaultContext"]    #(re-matches context-id-regex (get % "id"))
+             ["f:previous"]          #(or (nil? %)
+                                          (re-matches commit-id-regex (get % "id")))
+             ["f:time"]              pos-int?
+             ["f:v"]                 nat-int?
+             ["f:data" "f:address"]  #(re-matches mem-addr-regex %)
+             ["f:data" "f:assert"]   #(and (vector? %) (every? map? %))
+             ["f:data" "f:retract"]  #(and (vector? %) (every? map? %))
+             ["f:data" "f:flakes"]   pos-int?
+             ["f:data" "f:size"]     pos-int?
+             ["f:data" "f:t"]        pos-int?
+             ["f:data" "f:previous"] #(or (nil? %)
+                                          (re-matches db-id-regex (get % "id")))}]
+        (is (every? #(every? (fn [[kp valid?]]
```

Comment:
Perhaps you could use the built-in [clojure.test `are` macro](https://clojuredocs.org/clojure.test/are) for this.

Reply:
Yeah, I started with that at one point. There's probably still a better refactoring of this involving that but it wasn't jumping out at me. The wrinkle was the `(every? #(every? ...))` construction and being careful not to make assertions I didn't mean (e.g. it's easy to accidentally make the assertion that every key in the actual is one of the keys in an expected set using the set-as-pred pattern; but that will cause empty / strict-subset map actuals to pass).

Reply:
I'm generally not a fan of `are` because if it fails it won't tell you which case actually failed. It's more laborious to use `is` for every case, but working with it later is a lot easier.